### PR TITLE
Add support to Keychron Q0

### PR DESCRIPTION
**Pull request by adophoxia**
## Description
Add support to Keychron Q0

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q0/q0.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    } break;
+                }
+            }
+            return false;
+#    endif
+    }
+    return true;
+}
+
+__attribute__((weak)) void rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
```
Comment (drashna) [inline keyboards/keychron/q0/q0.c:45]:
```suggestion
__attribute__((weak)) void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
```

File: keyboards/keychron/q0/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+* Hardware Availability: [Keychron](https://www.keychron.com)
+
+Make example for this keyboard (after setting up your build environment):
+  ```
+    make keychron/q0/q0_rev_0130:default
+  ```
+Flashing example for this keyboard:
+  ```
+    make keychron/q0/q0_rev_0130:default:flash
+  ```
```
Comment (drashna) [inline keyboards/keychron/q0/readme.md:16]:
```suggestion

    make keychron/q0/q0_rev_0130:default

Flashing example for this keyboard:

    make keychron/q0/q0_rev_0130:default:flash

```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/keychron/q0/config.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+// #define RGB_MATRIX_KEYPRESSES
+
+/* Enable receive custom command from host */
+#define RAW_HID_CMD 0xAB
```
Comment (zvecr) [inline keyboards/keychron/q0/config.h:105]:
```suggestion
```

File: keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+enum layers { _BASE, _FN1, _RESERVED1, _RESERVED2 };
```
Comment (zvecr) [inline keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c:19]:
```suggestion
enum layers { _BASE, _FN1 };
```

File: keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        _______,    _______, _______,
+        _______,             _______,   _______),
+
+    [_RESERVED2] = LAYOUT_numpad_6x4(
+        _______,    _______, _______,   _______,
+        _______,    _______, _______,   _______,
+        _______,    _______, _______,
+        _______,    _______, _______,   _______,
+        _______,    _______, _______,
+        _______,             _______,   _______)
```
Comment (zvecr) [inline keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c:52]:
```suggestion
        RGB_TOG,             KC_MNXT,   _______)
```

File: keyboards/keychron/q0/config.h [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+/* NKRO */
+#define FORCE_NKRO
+
+/* turn off effects when suspended */
+#define RGB_DISABLE_WHEN_USB_SUSPENDED
+
+/* We have 2KB EEPROM size on STM32L432 */
+#define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 2047
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
```
Comment (zvecr) [inline keyboards/keychron/q0/config.h:48]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q0/config.h (hunk 6 above)
Comment (adophoxia) [reply keyboards/keychron/q0/config.h:48]: Would I have to include the 2 lines for the wear leveling algorithm needed for the 4KB of Emulated EEPROM, or is what is there rn fine?

File: keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c (hunk 5 above)
Comment (adophoxia) [reply keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c]: Removed 2 Reserved Keymap layers.

File: keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c (hunk 4 above)
Comment (adophoxia) [reply keyboards/keychron/q0/rev_0130/keymaps/default/keymap.c]: Removed 2 keymap layers from enum

File: keyboards/keychron/q0/config.h (hunk 3 above)
Comment (adophoxia) [reply keyboards/keychron/q0/config.h]: Removed RAW_HID

**Comment by adophoxia** [comment]
Ty for merging, @drashna 👍

**Pull request merged**